tests/component/sections/edit-collection/EditCollection.spec.tsx

```diff
@@ -0,0 +1,146 @@
+const userRepository: UserRepository = {} as UserRepository
+
+describe('CreateCollection', () => {
+  beforeEach(() => {
+    collectionRepository.create = cy.stub().resolves(1)
+    collectionRepository.getById = cy.stub().resolves(collection)
+    collectionRepository.getUserPermissions = cy.stub().resolves(userPermissionsMock)
+    collectionRepository.getFacets = cy.stub().resolves(collectionFacetsMock)
+    metadataBlockInfoRepository.getByCollectionId = cy.stub().resolves(collectionMetadataBlocksInfo)
+    metadataBlockInfoRepository.getAll = cy.stub().resolves(allMetadataBlocksMock)
+    metadataBlockInfoRepository.getAllFacetableMetadataFields = cy
+      .stub()
+      .resolves(allFacetableMetadataFieldsMock)
+    userRepository.getAuthenticated = cy.stub().resolves(testUser)
+  })
+
+  it('should show loading skeleton while loading the owner collection', () => {
+    const DELAYED_TIME = 200
+    collectionRepository.getById = cy.stub().callsFake(() => {
+      return Cypress.Promise.delay(DELAYED_TIME).then(() => collection)
+    })
+
+    cy.customMount(
+      <EditCollection
+        collectionId="root"
+        collectionRepository={collectionRepository}
+        metadataBlockInfoRepository={metadataBlockInfoRepository}
+      />
+    )
+    cy.clock()
+
+    cy.findByTestId('edit-collection-skeleton').should('exist')
+
+    cy.tick(DELAYED_TIME)
+
+    cy.findByTestId('edit-collection-skeleton').should('not.exist')
+
+    cy.clock().then((clock) => clock.restore())
+  })
+
+  it('should render the correct breadcrumbs', () => {
+    cy.customMount(
+      <EditCollection
+        collectionId="root"
+        collectionRepository={collectionRepository}
+        metadataBlockInfoRepository={metadataBlockInfoRepository}
+      />
+    )
+
+    cy.findByRole('link', { name: 'Root' }).should('exist')
+
+    cy.get('li[aria-current="page"]')
+      .should('exist')
+      .should('have.text', 'Edit Collection')
+      .should('have.class', 'active')
+  })
+
+  it('should show page not found when collection does not exist', () => {
+    collectionRepository.getById = cy.stub().resolves(null)
+
+    cy.customMount(
+      <EditCollection
+        collectionId="root"
+        collectionRepository={collectionRepository}
+        metadataBlockInfoRepository={metadataBlockInfoRepository}
+      />
+    )
+
+    cy.findByText('Page Not Found').should('exist')
+  })
+
+  it('should show alert error message when user is not allowed to edit the collection', () => {
+    collectionRepository.getUserPermissions = cy.stub().resolves(
+      CollectionMother.createUserPermissions({
+        canEditCollection: false
+      })
+    )
+
+    cy.mountAuthenticated(
+      <EditCollection
+        collectionId="root"
+        collectionRepository={collectionRepository}
+        metadataBlockInfoRepository={metadataBlockInfoRepository}
+      />
+    )
+    cy.findAllByTestId('not-allowed-to-edit-collection-alert').should('exist')
```

Comment:
For testing the appearance of text on the page, I usually test for the existence of the text itself, rather than a testId.  Is there a rule of thumb that you use to check for the id rather than the text?

Reply:
changed to `findByText`